tool_path_planner/src/surface_walk_raster_generator.cpp

```diff
@@ -1198,94 +1198,66 @@ vtkSmartPointer<vtkPolyData> SurfaceWalkRasterGenerator::createStartCurve()
   vtkSmartPointer<vtkPoints> pts = vtkSmartPointer<vtkPoints>::New();
   line->SetPoints(pts);
-  double corner[3];
-  double max[3];
-  double mid[3];
-  double min[3];
-  double size[3];
 
-  if (config_.cut_direction[0] || config_.cut_direction[1] || config_.cut_direction[2])
-  {
-    max[0] = config_.cut_direction[0];
-    max[1] = config_.cut_direction[1];
-    max[2] = config_.cut_direction[2];
-    avg_center[0] = config_.cut_centroid[0];
-    avg_center[1] = config_.cut_centroid[1];
-    avg_center[2] = config_.cut_centroid[2];
-  }
-  else
+  Eigen::Vector3d max, min;
   {
+    // Calculate the object-oriented bounding box to determine how wide a cutting plane should be
     vtkSmartPointer<vtkOBBTree> obb_tree = vtkSmartPointer<vtkOBBTree>::New();
     obb_tree->SetTolerance(0.001);
     obb_tree->SetLazyEvaluation(0);
-    obb_tree->ComputeOBB(mesh_data_->GetPoints(), corner, max, mid, min, size);
-
-    // size gives the length of each vector (max, mid, min) in order, normalize the size vector by the max size for
-    // comparison
-    double m = size[0];
-    size[0] /= m;
-    size[1] /= m;
-    size[2] /= m;
-
-    // ComputeOBB uses PCA to find the principle axes, thus for square objects it returns the diagonals instead
-    // of the minimum bounding box.  Compare the first and second axes to see if they are within 1% of each other
-    if (size[0] - size[1] < 0.01)
+    Eigen::Vector3d mid, corner, norm_size;
+    obb_tree->ComputeOBB(mesh_data_->GetPoints(), corner.data(), max.data(), mid.data(), min.data(), norm_size.data());
+
+    // Use the specified cut direction if it is not [0, 0, 0]
+    Eigen::Vector3d cut_dir(config_.cut_direction);
+    if (!cut_dir.isApprox(Eigen::Vector3d::Zero()))
+    {
+      max = cut_dir.normalized() * max.norm();
+    }
+    else
     {
-      // if object is square, need to average max and mid in order to get the correct axes of the object
-      m = sqrt(max[0] * max[0] + max[1] * max[1] + max[2] * max[2]);
-      double temp_max = m;
-      max[0] /= m;
-      max[1] /= m;
-      max[2] /= m;
-      m = sqrt(mid[0] * mid[0] + mid[1] * mid[1] + mid[2] * mid[2]);
-      mid[0] /= m;
-      mid[1] /= m;
-      mid[2] /= m;
-
-      max[0] = (max[0] + mid[0]) * temp_max;
-      max[1] = (max[1] + mid[1]) * temp_max;
-      max[2] = (max[2] + mid[2]) * temp_max;
+      // ComputeOBB uses PCA to find the principle axes, thus for square objects it returns the diagonals instead
+      // of the minimum bounding box.  Compare the first and second axes to see if they are within 1% of each other
+      if (norm_size[1] / norm_size[0] > 0.99)
+      {
+        // if object is square, need to average max and mid in order to get the correct axes of the object
+        max = (max.normalized() + mid.normalized()) * max.norm();
+      }
     }
   }
+
   // Create additional points for the starting curve
-  double pt[3];
-  double raster_axis[3];
+  Eigen::Vector3d raster_axis;
```

Comment:
Should this be initialized? I noticed an else below that has not content so it may be possible to use this when it is contains uninitialized data.

Reply:
Good catch. Technically this retains the original behavior, but the original behavior would have an issue if the `raster_wrt_global_axes` was defined